public/game-master/view-models/app-viewmodel.coffee

```diff
@@ -48,6 +49,13 @@ class AppViewModel
 					solution
 			]
 
+		@current_puzzle_visible_soultion = new nx.Cell
```

Comment:
`current_puzzle_solution` could be. But the solution exists always but it's visible not in all cases. That's why there's a prefix `visible`